### PR TITLE
More coverage in fuzz-json

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/shared/json.c [hunk 1]
```diff
@@ -4499,14 +4489,8 @@ int json_dispatch_strv(const char *name, JsonVariant *variant, JsonDispatchFlags
...
-        JsonVariant **p = userdata;
-
-        assert(variant);
-        assert(p);
-
-        json_variant_unref(*p);
-        *p = json_variant_ref(variant);
-
+        JsonVariant **p = ASSERT_PTR(userdata);
+        JSON_VARIANT_REPLACE(*p, ASSERT_PTR(variant));
```
Comment (yuwata) [inline src/shared/json.c]: missing `json_variant_ref()`.

**Review by yuwata — CHANGES_REQUESTED** [review]
LGTM. But one issue.

File: src/shared/json.c (hunk 1 above)
Comment (keszybz) [reply src/shared/json.c]: Eh, indeed. I tried a few different forms here and apparently managed to introduce the mistake on the way.

**Comment by keszybz** [comment]
> Though I am not sure it is related to the above point, CIs are unhappy.

Yeah, I'm not sure either. I pushed without any other changes for now.

**Pull request merged**